next-auth-with-directus/index.md

```diff
@@ -0,0 +1,1040 @@
+    } catch (e: any) {
+      console.log(e)
+        setError('The reset password token is invalid, please request for a new password reset link!')
+    }
+  };
+  return (
+    <form onSubmit={handleFormSubmit} className=" md:w-[500px] w-full">
+      <h1 className="capitalize text-lg font-medium my-4">
+        Provide a new password for your account
+      </h1>
+      {success &&
+        <p className='text-green-600 bg-green-50 p-2 rounded-md'>
+          {success}
+        </p>}
+      {error &&
+        <p className='text-red-600 bg-red-50 p-2 rounded-md'>
+          {error}
+        </p>}
+      <p>
+       Enter your new password for your account
+      </p>
+      <input
+        type="password"
+        placeholder="Enter your new password"
+        name="password"
+        required
+        onChange={(e) => setNewPassword(e.target.value)}
+        autoComplete='new-password'
+        className="py-2 px-4 border border-indigo-500 rounded-md w-full my-5"
+      />
+      <button className="bg-indigo-600 text-white py-3 px-6 w-full rounded-md">
+        Create new password
+      </button>
+    </form>
+  );
+}
+ ```
+
+- The _reset-password/form.tsx_ accepts a token and sends a request to Directus using the Directus SDK with the `token` and `newPassword` as parameters for changing the user's password.
+
+- If this request is successful, it redirects the user to the login page to log in with their new password.
+
+Inside of the _page.tsx_, update the content to be:
+
+`app/reset-password/page.tsx`
+
+```tsx
+import { redirect } from 'next/navigation';
+import ResetPasswordForm from './form';
+
+
+export default async function ResetPasswordPage({ searchParams }: { searchParams: { token: string } }) {
+  const { token } = searchParams
+  if (!token)
+    redirect('/login')
+  return (
+    <div className="flex h-full justify-center flex-col min-h-screen items-center md:p-24 p-5">
+      <ResetPasswordForm  token={token} />
+    </div>
+  );
+}
+```
+
+The _page.tsx_ components checks if a token is present in the `reset-password` url; if it is present, it displays the `ResetPasswordPage`. Otherwise, it redirects the user to the login page.
+
+## Summary
+
+That's a wrap 🥳; in this article, you've successfully built an authentication system with password reset functionality using `Nextjs`, `NextAuth`, and `Directus`.
+This is just a glimpse of what you can implement with Directus
+Directus runs entirely as a backend service, meaning you can build complex backend services that will serve your frontend application with any database of your choice
+
+Some possible steps you can consider to improve this application:
+
+- Improving the functionality of the authentication system to accept OAuth providers like `Google`, `Twitter`, et. c
+- Improving error handling to show the correct errors to your users properly
+- Add a
+- Create new `Item` models that your `Customers` can use to create their data from your frontend application
+
```

Comment:
Add the note here about customizing the `directus_users` data model to contain extra fields